cli/src/codecs.js

```diff
@@ -0,0 +1,363 @@
+    enc: () => instantiateEmscriptenWasm(webpEnc, webpEncWasm),
+    defaultEncoderOptions: {
+      quality: 75,
+      target_size: 0,
+      target_PSNR: 0,
+      method: 4,
+      sns_strength: 50,
+      filter_strength: 60,
+      filter_sharpness: 0,
+      filter_type: 1,
+      partitions: 0,
+      segments: 4,
+      pass: 1,
+      show_compressed: 0,
+      preprocessing: 0,
+      autofilter: 0,
+      partition_limit: 0,
+      alpha_compression: 1,
+      alpha_filtering: 1,
+      alpha_quality: 100,
+      lossless: 0,
+      exact: 0,
+      image_hint: 0,
+      emulate_jpeg_size: 0,
+      thread_level: 0,
+      low_memory: 0,
+      near_lossless: 100,
+      use_delta_palette: 0,
+      use_sharp_yuv: 0,
+    },
+    autoOptimize: {
+      option: 'quality',
+      min: 0,
+      max: 100,
+    },
+  },
+  avif: {
+    name: 'AVIF',
+    extension: 'avif',
+    detectors: [/^\x00\x00\x00 ftypavif\x00\x00\x00\x00/],
+    dec: () => instantiateEmscriptenWasm(avifDec, avifDecWasm),
+    enc: () => instantiateEmscriptenWasm(avifEnc, avifEncWasm),
+    defaultEncoderOptions: {
+      minQuantizer: 33,
+      maxQuantizer: 63,
+      minQuantizerAlpha: 33,
+      maxQuantizerAlpha: 63,
+      tileColsLog2: 0,
+      tileRowsLog2: 0,
+      speed: 8,
+      subsample: 1,
+    },
+    autoOptimize: {
+      option: 'maxQuantizer',
+      min: 0,
+      max: 62,
+    },
+  },
+  jxl: {
+    name: 'JPEG-XL',
+    extension: 'jxl',
+    detectors: [/^\xff\x0a/],
+    dec: () => instantiateEmscriptenWasm(jxlDec, jxlDecWasm),
+    enc: () => instantiateEmscriptenWasm(jxlEnc, jxlEncWasm),
+    defaultEncoderOptions: {
+      speed: 4,
+      quality: 75,
+      progressive: false,
+      epf: -1,
+      nearLossless: 0,
+      lossyPalette: false,
+    },
+    autoOptimize: {
+      option: 'quality',
+      min: 0,
+      max: 100,
+    },
+  },
+  wp2: {
+    name: 'WebP2',
+    extension: 'wp2',
+    detectors: [/^\xF4\xFF\x6F/],
+    dec: () => instantiateEmscriptenWasm(wp2Dec, wp2DecWasm),
+    enc: () => instantiateEmscriptenWasm(wp2Enc, wp2EncWasm),
+    defaultEncoderOptions: {
+      quality: 75,
+      alpha_quality: 75,
+      effort: 5,
+      pass: 1,
+      sns: 50,
+      uv_mode: 0 /*UVMode.UVModeAuto*/,
+      csp_type: 0 /*Csp.kYCoCg*/,
+      error_diffusion: 0,
+      use_random_matrix: false,
+    },
+    autoOptimize: {
+      option: 'quality',
+      min: 0,
+      max: 100,
+    },
+  },
+  oxipng: {
+    name: 'OxiPNG',
+    extension: 'png',
+    detectors: [/^\x89PNG\x0D\x0A\x1A\x0A/],
+    dec: async () => {
+      await pngEncDecPromise;
+      return { decode: pngEncDec.decode };
+    },
+    enc: async () => {
+      await pngEncDecPromise;
+      await oxipngPromise;
+      return {
+        encode: (buffer, width, height, opts) => {
+          const simplePng = new Uint8Array(
+            pngEncDec.encode(new Uint8Array(buffer), width, height),
```

Comment:
```suggestion
            pngEncDec.encode(buffer, width, height),
```

Also, wasm-bindgen returns a `Vec<u8>` so this stuff is already owned by JS - no need to reclone.

Reply:
Er, wait, wrong suggestion.